### PR TITLE
Multiple small fixes for i18n issues

### LINKED ISSUE #6464: Make sure all strings can be translated
### Describe the bug
While working on https://github.com/Automattic/woocommerce-payments/pull/6388, we discovered there are many other strings that we need to wrap in `__(` `)` to make these strings ready for translations.

The work has already started in https://github.com/Automattic/woocommerce-payments/pull/6385 but we'll need to continue and make this a team effort.

### PR DESCRIPTION
**Pull request by elazzabi**
Fixes #6464 

This is a continuation of https://github.com/Automattic/woocommerce-payments/pull/6385. The older PR was behind the main branch and it was a pain to rebase it with all the new changes. This is a fresh start with cherry-picks from the older branch.

#### Changes proposed in this Pull Request
- Making sure most phrases are ready to be translated by wrapping them in our i18n helpers.
- Making sure translator comments are included with each translation
- **NEW**: Adds a mention to check i18n string to the new PR template

#### Testing instructions

* Make sure all changes look good
* Run `npm run build`
* Open `languages/woocommerce-payments.pot` and make sure the new phrases are showing correctly. (It's easy to copy a phrase and searching for it)

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8181` or branch name `fix/i18n-for-plugin` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/i18n-for-plugin) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 0e1195ec456e19001c03a1edc9f0d4d538328a61
- Build time: 2024-02-20 11:57:43 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (elazzabi) [inline .github/PULL_REQUEST_TEMPLATE.md]: This is an extra check added to the PR description for future PRs

File: client/payment-details/summary/index.tsx
Comment (mordeth) [inline client/payment-details/summary/index.tsx]: The `sprintf` is not utilized in the file, it should be removed.

**Review by mordeth — APPROVED** [review]
✅  Code changes looking good!

Added a minor linting suggestion in the comments.

File: client/payment-details/summary/index.tsx
Comment (elazzabi) [reply client/payment-details/summary/index.tsx]: Indeed, fixed 85e7afefb

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (anu-rock) [reply .github/PULL_REQUEST_TEMPLATE.md]: Thanks for adding this :)

However, since this change will affect all future PRs I think it's good to surface this proposed change in the wider group to check whether there's any resistance. In case of no objections, we'll go ahead with this addition.

File: changelog/i18n-fixes
Comment (anu-rock) [inline changelog/i18n-fixes]: I've always found changelog written in the present tense - just like commit messages - easier to read. On a quick glance, that seems to be the case for most changelogs [here](https://github.com/Automattic/woocommerce-payments/blob/develop/changelog.txt).

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (anu-rock) [reply .github/PULL_REQUEST_TEMPLATE.md]: Also, I think adding a link to a page that explains what "ready for i18n" looks like would be useful.

File: client/overview/modal/reset-account/index.tsx
Comment (anu-rock) [inline client/overview/modal/reset-account/index.tsx:25]: Good catch 👍

File: includes/admin/class-wc-rest-payments-orders-controller.php
Comment (anu-rock) [inline includes/admin/class-wc-rest-payments-orders-controller.php]: Are these user-facing error messages? if not, I wonder if there's any value in translating them 🤔

File: includes/class-wc-payment-gateway-wcpay.php
Comment (anu-rock) [inline includes/class-wc-payment-gateway-wcpay.php]: Same comment as for exceptions.

**Review by anu-rock — COMMENTED** [review]
Wow, you've caught so many untranslated strings. Thanks for taking the time to tidy them up. I've left a few comments, happy to do another round soon.

___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | Not applicable |
| Naming  | Not applicable |
| Typing  | Not applicable |
| Code comments  | ✅ All good |
| Changelog  | 👍  Almost good (left a comment) |
| Testing instructions  | ✅ All good |
| Demo (video or JN site)  | Not applicable |
| Manual testing  | Skipped |

File: includes/admin/class-wc-rest-payments-orders-controller.php
Comment (elazzabi) [reply includes/admin/class-wc-rest-payments-orders-controller.php]: I had the same question when I was working on the changes.

My initial thought is for developers to debug the problems in the code (and in their language). But if this is not something we see often, we can revert those changes.

I agree that our priority should be user-facing strings of course.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (elazzabi) [reply .github/PULL_REQUEST_TEMPLATE.md]: What do you think about linking to the [package's README](https://github.com/WordPress/gutenberg/tree/trunk/packages/i18n)?

I was planning to P2 this change after merging it, but I believe it should be before as you suggested. Will do it 👍

File: changelog/i18n-fixes
Comment (elazzabi) [reply changelog/i18n-fixes]: Indeed. Fixed c82e40524677d906862d136f9cd405f47352b089

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (elazzabi) [reply .github/PULL_REQUEST_TEMPLATE.md]: Created a P2 to ask for feedback. The summary is here paJDYF-cz6-p2#comment-23249

I will remove the checkbox for now and move forward with the changes.

File: includes/admin/class-wc-rest-payments-orders-controller.php
Comment (anu-rock) [reply includes/admin/class-wc-rest-payments-orders-controller.php]: > My initial thought is for developers to debug the problems in the code (and in their language).

That is a kind thought, Ahmed. However, it is my opinion that developers are a small, controlled group of users that usually work well with the English language. I am not sure if there's a precedent for translating developer-facing error messages, but we could check how, say, Calypso handles it. Either way, I feel that the benefit of having debug messages in developers' native tongues may not offset the effort/cost of translating them.

File: changelog/i18n-fixes
Comment (anu-rock) [reply changelog/i18n-fixes]: Thank you, it does read better now ❤️ But I feel there's still scope to expand on what "i18n instances" means since internationalization is a broader term.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (anu-rock) [reply .github/PULL_REQUEST_TEMPLATE.md]: Thanks for starting a wider discussion. Marking this thread resolved while we discuss a solution on P2.

File: includes/admin/class-wc-rest-payments-orders-controller.php
Comment (mordeth) [reply includes/admin/class-wc-rest-payments-orders-controller.php]: I agree with Anurag. Translating error messages doesn't provide any value and can potentially cause confusion during debugging.

Comment (elazzabi) [reply includes/admin/class-wc-rest-payments-orders-controller.php]: Reverted in 3cef9c41497c6921f190e436cc0aa6aa93f63a81 and 4f76bd73e6c47f7eb9f89c9c4bebc689c93896fe

File: changelog/i18n-fixes
Comment (elazzabi) [reply changelog/i18n-fixes]: Changed to "Fix a few untranslated phrases in the plugin"

**Review by anu-rock — APPROVED** [review]
LGTM after the latest changes. All open comments are resolved as well. Let's go, Rambo 🚀

**Pull request merged**
